### PR TITLE
Add add compiler_rt routines for float to signed integer conversion

### PR DESCRIPTION
**Pull request by winksaville**
The routines are:

  __fixdfdi, __fixdfsi, __fixdfti,
  __fixsfdi, __fixsfsi, __fixsfti,
  __fixtfdi, __fixtfsi, __fixtfti.

These all call fixint which is a generic zig function that does the
conversion:

  pub fn fixint(comptime fp_t: type, comptime fixint_t: type, a: fp_t) fixint_t

There are also a set tests:

  __fixdfdi_test, __fixdfsi_test, __fixdfti_test,
  __fixsfdi_test, __fixsfsi_test, __fixsfti_test,
  __fixtfdi_test, __fixtfsi_test, __fixtfti_test.

There is also fixint_test.zig which does some basic tests
including tests small integers i1, i2, i3.

### CODE REVIEW

**Comment by winksaville** [comment]
Txs

**Comment by winksaville** [comment]
How do I start a CI build after adding a new commit?

**Comment by winksaville** [comment]
Close and reopen to see if that will kickoff a CI build

**Comment by winksaville** [comment]
Reopne

**Comment by winksaville** [comment]
@andrewrk, anything addional you'd like me to do on this? I assume you'd like it to be squashed, but want to wait until you approve it.

**Comment by andrewrk** [comment]
Hi @winksaville, thanks for doing this work! I've been visiting my family for holiday travel and I have a lot of pull requests to review, including this one. I can get to a couple tomorrow, and then I'll have a full day Wednesday to make sure everybody's code gets looked at and merged.

File: std/special/compiler_rt/fixtfsi_test.zig [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+const assert = std.debug.assert;
+const warn = std.debug.warn;
+
+fn test__fixtfsi(a: f128, expected: i32) void {
+    const x = __fixtfsi(a);
+    //warn("a={}:{x} x={}:{x} expected={}:{x}:u32({x})\n", a, @bitCast(u128, a), x, x, expected, expected, @bitCast(u32, expected));
+    assert(x == expected);
+}
+
+const f128_max = @bitCast(f128, u128(0x7FFEFFFFFFFFFFFFFFFFFFFFFFFFFFFF)); // TODO: Remove when std.math.f128_max defined
```
Comment (andrewrk) [inline std/special/compiler_rt/fixtfsi_test.zig]: Want to go ahead and move these to `std.math`?

**Comment by winksaville** [comment]
Close and reopen to kick off build

**Comment by winksaville** [comment]
Reopen

**Pull request merged**